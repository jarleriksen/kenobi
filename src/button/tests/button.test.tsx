import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Button } from '..';

describe('Button', () => {
  test('it renders correctly', () => {
    const { getByTestId, container, asFragment } = render(
      <Button>Hello!</Button>,
    );

    const button = getByTestId('kenobi-button');

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('Hello!');
    expect(asFragment()).toMatchSnapshot();
  });

  test('it renders as an anchor tag', () => {
    const { getByTestId, container, asFragment } = render(
      <Button isLink={true} href="https://example.com">
        Hello!
      </Button>,
    );

    const button = container.getElementsByTagName('a')[0];

    expect(button).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('it renders icon before', () => {
    const { getByTestId, container, asFragment } = render(
      <Button iconBefore="edit">Hello!</Button>,
    );

    const button = getByTestId('kenobi-button');
    const icon = getByTestId('kenobi-icon');

    expect(button).toContainElement(icon);
    expect(asFragment()).toMatchSnapshot();
  });

  test('it renders icon after', () => {
    const { getByTestId, container, asFragment } = render(
      <Button iconAfter="edit">Hello!</Button>,
    );

    const button = getByTestId('kenobi-button');
    const icon = getByTestId('kenobi-icon');

    expect(button).toContainElement(icon);
    expect(asFragment()).toMatchSnapshot();
  });

  test('it renders spinner', () => {
    const { getByTestId, container, asFragment } = render(
      <Button isLoading={true}>Hello!</Button>,
    );

    const button = getByTestId('kenobi-button');
    const spinner = getByTestId('kenobi-spinner');

    expect(button).toContainElement(spinner);
    expect(asFragment()).toMatchSnapshot();
  });
});
