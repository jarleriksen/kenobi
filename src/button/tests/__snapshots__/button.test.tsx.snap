// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it renders as an anchor tag 1`] = `
<DocumentFragment>
  <a
    class="sc-bwzfXH fokvxT"
    data-testid="kenobi-button"
    href="https://example.com"
  >
    <span
      class="sc-htpNat jeMNHo"
    >
      Hello!
    </span>
  </a>
</DocumentFragment>
`;

exports[`Button it renders correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-bwzfXH fokvxT"
    data-testid="kenobi-button"
  >
    <span
      class="sc-htpNat jeMNHo"
    >
      Hello!
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button it renders icon after 1`] = `
<DocumentFragment>
  <button
    class="sc-bwzfXH fokvxT"
    data-testid="kenobi-button"
  >
    <span
      class="sc-htpNat jeMNHo"
    >
      Hello!
      <svg
        data-icon="edit"
        data-testid="kenobi-icon"
        height="12"
        style="margin-left: 8px; margin-right: -3px; fill: white;"
        viewBox="0 0 16 16"
        width="12"
      >
        <path
          d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button it renders icon before 1`] = `
<DocumentFragment>
  <button
    class="sc-bwzfXH fokvxT"
    data-testid="kenobi-button"
  >
    <span
      class="sc-htpNat jeMNHo"
    >
      <svg
        data-icon="edit"
        data-testid="kenobi-icon"
        height="12"
        style="margin-left: -3px; margin-right: 8px; fill: white;"
        viewBox="0 0 16 16"
        width="12"
      >
        <path
          d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
          fill-rule="evenodd"
        />
      </svg>
      Hello!
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button it renders spinner 1`] = `
<DocumentFragment>
  <button
    class="sc-bwzfXH fokvxT"
    data-testid="kenobi-button"
    disabled=""
  >
    <div
      class="sc-bdVaJa tuPFw"
      data-testid="kenobi-spinner"
      size="16"
    />
    <span
      class="sc-htpNat bqYOtN"
    >
      Hello!
    </span>
  </button>
</DocumentFragment>
`;
